### DISCUSSION TITLE
Batch email notifications per week, month, etc.

### DISCUSSION THREAD

**Discussion by hsandt** [community/community#12504] [Projects and Issues]
**Context**

I'm subscribed to a very active thread and receive several comment notification emails every day about it. It is interesting so I don't want to unsubscribe, just receive notifications less often.

**Proposal**

It would be nice if we could have a setting to wait for a certain period of time or amount of activity before receiving a notification email. Then, the notification email could either contain all activity content (if not too long), or just a link to the relevant page, so the user can take their time to read through (in the case of comments, it's more convenient to read 10 comments directly on the page, so the link could send the user to the first unread/new comment since last time).

We could work with fixed day time / weekday, or with a sliding time window.

Example of possible notification settings:
* notify every activity (current behavior)
* notify daily (end of day or custom selected day time)
* notify weekly (fixed day)
* notify weekly (window of 7 days after the first new comment)
* notify 7 days after the first new comment OR at the end of the first day (or first N days) without any new comment (it’s just a trick to identify running discussions and only notify them after a break, or if they have been running for long and don't seem to stop any time soon, so you can just join in)
* notify monthly
* notify after a window of 30 days
* any variant tweaking the number of hours/days to wait

Additional settings (thanks to [madhavajay](https://github.community/u/madhavajay)):
* Filter and/or group email notifications by categories such as “PRs”, “Issues”, “Tests”, etc. (currently, we can decide which category to follow on a given repository with https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#configuring-your-watch-settings-for-an-individual-repository but that applies to both email notifications and the [notifications inbox](https://github.com/notifications?query=is%3Aunread); we may want to filter out some categories for emails only)

**Related issues**

This is a reboot of https://github.community/t/cannot-batch-email-notifications-per-week-month/153283 (since we should now use this repository for feedback, not GitHub community) and https://github.com/isaacs/github/issues/340 (a variant specialized on PR that also suggests letting the reviewer decide when to send the batch notifications).

Also interesting ideas in https://github.com/github/feedback/discussions/10869 which suggests showing the average activity on a given thread before you subscribe. It could help us select an ideal batch frequency (the interface could auto-select a good frequency too).

**Workaround**

Unsubscribe from all email notifications, and add a weekly reminder on your computer to check the [notifications inbox](https://github.com/notifications?query=is%3Aunread) on the website manually... Its interface allows filtering by category, unread, etc. and much easier navigation, where you open the thread at the first new message by clicking on the +X text. It's good if you need to check updates regularly, but if updates are irregular and you need to be aware of the latest updates at least every N days, then you still need *some* email to tell you that something has happened every N days.

**Comment by marc-hb** [comment]
If/when this gets implemented in some shape or form, the default "cooldown" value should be changed to 3 minutes for Pull Requests. Absolutely no one wants to receive 10 notification emails in 10 minutes with one review comment in each - which is how the system works now :-(
In other words, emails should be sent only after 3 idle minutes by default. Many other systems get this right.

In Pull Requests it's possible to mitigate a bit this spam problem by reminding users that the green, "Start Review" button:
- batches notifications
- is not just for reviewers for but for authors too (frequent misunderstanding in my experience).

Batching is also great when you realize that some review comments cross-reference or duplicate each other, this gives a chance to "clean-up" / proof-read before sending.

It's only a mitigation: some users forget or don't care or don't like "Start Review" because they may forget to submit when they're done.

**Comment by wizseek** [comment]
Surprised this feature is not being requested more often. It is close to impossible to go through all the email given the current setting.

**Reply by RobertTheBlair** [reply]
My org just was switched from BitBucket to Github, and the experience around notifications has been terrible.
Please steal that feature from BitBucket.  Specifically, batching emails as a receiver notification preference, rather than relying on the commenter to remember

**Comment by bandtank** [comment]
Any updates? The lack of control is really burdensome.
